dspace-api/src/main/java/org/dspace/importer/external/arxiv/metadatamapping/contributor/ArXivIdMetadataContributor.java

```diff
@@ -0,0 +1,187 @@
+     * @param field
+     * <a href="https://github.com/DSpace/DSpace/tree/master/dspace-api/src/main/java/org/dspace/importer/external#metadata-mapping-">MetadataFieldConfig</a>
+     */
+    public ArXivIdMetadataContributor(String query, Map<String, String> prefixToNamespaceMapping,
+                                           MetadataFieldConfig field) {
+        this.query = query;
+        this.prefixToNamespaceMapping = prefixToNamespaceMapping;
+        this.field = field;
+    }
+
+    /**
+     * Empty constructor for ArXivIdMetadataContributor
+     */
+    public ArXivIdMetadataContributor() {
+
+    }
+
+    private String query;
+
+    /**
+     * Return the MetadataFieldConfig used while retrieving MetadatumDTO
+     *
+     * @return MetadataFieldConfig
+     */
+    public MetadataFieldConfig getField() {
+        return field;
+    }
+
+    /**
+     * Setting the MetadataFieldConfig
+     *
+     * @param field MetadataFieldConfig used while retrieving MetadatumDTO
+     */
+    @Required
+    public void setField(MetadataFieldConfig field) {
+        this.field = field;
+    }
+
+    /**
+     * Return query used to create an xpathExpression on, this query is used to
+     *
+     * @return the query this instance is based on
+     */
+    public String getQuery() {
+        return query;
+    }
+
+    @Required
+    public void setQuery(String query) {
+        this.query = query;
+    }
+
+    /**
+     * Retrieve the metadata associated with the given object.
+     * Depending on the retrieved node (using the query), different types of values will be added to the MetadatumDTO
+     * list
+     *
+     * @param t A class to retrieve metadata from.
+     * @return a collection of import records. Only the identifier of the found records may be put in the record.
+     */
+    @Override
+    public Collection<MetadatumDTO> contributeMetadata(OMElement t) {
+        List<MetadatumDTO> values = new LinkedList<>();
+        try {
+            AXIOMXPath xpath = new AXIOMXPath(query);
+            for (String ns : prefixToNamespaceMapping.keySet()) {
+                xpath.addNamespace(prefixToNamespaceMapping.get(ns), ns);
+            }
+            List<Object> nodes = xpath.selectNodes(t);
+            for (Object el : nodes) {
+                if (el instanceof OMElement) {
+                    values.add(metadataFieldMapping.toDCValue(field, ((OMElement) el).getText()));
+                } else if (el instanceof OMAttribute) {
+                    values.add(metadataFieldMapping.toDCValue(field, ((OMAttribute) el).getAttributeValue()));
+                } else if (el instanceof String) {
+                    values.add(metadataFieldMapping.toDCValue(field, (String) el));
+                } else if (el instanceof OMText) {
+                    values.add(metadataFieldMapping.toDCValue(field, ((OMText) el).getText()));
+                } else {
+                    System.err.println("node of type: " + el.getClass());
```

Comment:
Also just noticed this uses System.err.println, while log.error is preferred. This is also happening at https://github.com/DSpace/DSpace/blob/main/dspace-api/src/main/java/org/dspace/importer/external/metadatamapping/contributor/SimpleXpathMetadatumContributor.java#L160 which is why I didn't see it immediately. This should be a small change to improve